### PR TITLE
Start language server when extension activated

### PR DESCRIPTION
**Pull request by shanluo911**
[AB#346673](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/346673)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,11 @@
           ],
           "default": "",
           "description": "Microsoft employee or public contributor"
+        },
+        "docs-build.realTimeValidation.enable": {
```
Comment (928PJY) [inline package.json]: So we call them both `enable real-time validation`? can we give a different name?

File: src/extension.ts [hunk 2]
```diff
@@ -84,6 +85,10 @@ export async function activate(context: vscode.ExtensionContext): Promise<Extens
 
     const codeActionProvider = new CodeActionProvider();
 
+    // Start language server
+    const startServerTriggerObserver = new StartLanguageServerObserver(environmentController);
```
Comment (928PJY) [inline src/extension.ts]: how about rename it to `languageServerManager`?

File: package.json [hunk 3]
```diff
@@ -67,6 +67,11 @@
           ],
           "default": "",
           "description": "Microsoft employee or public contributor"
+        },
+        "docs-build.realTimeValidation.automaticallyEnable": {
+          "type": "boolean",
+          "default": false,
+          "description": "Automatically enable real-time validation"
```
Comment (928PJY) [inline package.json]: With this flag enabled, the real-time validation will be automatically enabled once the opened repository is a valid Docs repository.

File: src/common/docsEnvironmentController.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ export class DocsEnvironmentController implements EnvironmentController, vscode.
     private _docsRepoType: DocsRepoType;
     private _configurationChangeListener: vscode.Disposable;
     private _userType: UserType;
+    private _enableRealTimeValidation: boolean;
```
Comment (928PJY) [inline src/common/docsEnvironmentController.ts]: should this be consistent with the config name?

File: src/observers/languageServerManager.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { BaseEvent } from '../common/loggingEvents';
```
Comment (928PJY) [inline src/observers/languageServerManager.ts]: consider moving this file to the `build` (which will be renamed to `validation` in the future as we talked)

File: src/common/docsEnvironmentController.ts (hunk 4 above)
Comment (shanluo911) [reply src/common/docsEnvironmentController.ts]: Feel a little bit weird directly using automaticallyEnableRealTimeValidation here, changing to 'enableAutomaticRealTimeValidation' be better I think.

**Review by 928PJY — APPROVED** [review]

**Pull request merged**